### ISSUE TITLE
Custom loss with one of the features used and a condition

### ISSUE THREAD

**Issue by Lledune** [keras-team/keras#12703]
I'm trying to make a custom function for the deviance in Keras. 
Deviance is calculated as : 2 * (log(yTrue) - log(yPred)) 

The problem here is that my yTrue values are rare event count and therefore often equal to 0, resulting in a -inf error. 

The derivation of deviance for my specific case (poisson unscaled deviance) gives a solution to this : 

- If yTrue = 0, then deviance is : 2 * D * yPred where D is a feature of my data. 
- If yTrue !=0, then deviance is : 2 * D * (yTrue * ln(yTrue) - yTrue * ln(yPred) - yTrue + yPred

There is two problems here i encounter : 

- I need to choose the function according to the value of yPred
- I also need to pass D as argument to the loss function 

I made a first iteration of a loss function before derivating deviance, adding small values to yTrue when it is equal to 0 to prevent the -Inf. problems, but it gives wrong results for deviance so i have to change it. 

    def DevianceBis(y_true, y_pred):
    y_pred = KB.maximum(y_pred, 0.0 + KB.epsilon()) #make sure ypred is positive or ln(-x) = NAN
    return (KB.sqrt(KB.square( 2 * KB.log(y_true + KB.epsilon()) - KB.log(y_pred))))

I know I have to use keras backend functions to perform this but can't get much information on the subject as i need both a condition and a feature used as explanatory variable

Thanks in advance
